internal/provider/resource_keys.go

```diff
@@ -0,0 +1,245 @@
+package provider
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-framework-timeouts/resource/timeouts"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/retry"
+
+	"github.com/humanitec/humanitec-go-autogen"
+	"github.com/humanitec/humanitec-go-autogen/client"
+)
+
+// Ensure provider defined types fully satisfy framework interfaces
+var _ resource.Resource = &ResourceKey{}
+var _ resource.ResourceWithImportState = &ResourceKey{}
+
+var defaultKeysReadTimeout = 2 * time.Minute
+var defaultKeysDeleteTimeout = 2 * time.Minute
+
+func NewResourceKey() resource.Resource {
+	return &ResourceKey{}
+}
+
+// ResourceKey defines the resource implementation.
+type ResourceKey struct {
+	client *humanitec.Client
+	orgId  string
+}
+
+// OperatorKeyModel describes the key data model.
+type OperatorKeyModel struct {
+	ID          types.String   `tfsdk:"id"`
+	Key         types.String   `tfsdk:"key"`
+	CreatedAt   types.String   `tfsdk:"created_at"`
+	CreatedBy   types.String   `tfsdk:"created_by"`
+	ExpiredAt   types.String   `tfsdk:"expired_at"`
```

Comment:
For other resources we don't expose those status flags yet, could we remove them until somebody asks for them?